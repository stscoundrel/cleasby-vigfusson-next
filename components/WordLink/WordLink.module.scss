.link {
  display: block;
  max-width: 160px;
  padding: 2px 0;
  margin: 4px 0;
}