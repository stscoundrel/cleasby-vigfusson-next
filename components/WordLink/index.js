import Link from 'next/link'

export default function WordLink({ data }) {
  const { slug, word } = data

  return (
   <Link key={`link${slug}`} href={`/word/${slug}`}>
      <a>{word}</a>
    </Link>
  )
}
