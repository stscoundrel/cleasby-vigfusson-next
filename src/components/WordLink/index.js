import Link from 'next/link'
import styles from './WordLink.module.scss'

export default function WordLink({ data }) {
  const { slug, word } = data

  return (
   <Link key={`link${slug}`} href={`/word/${slug}`}>
      <a className={styles.link}>{word.toLowerCase()}</a>
    </Link>
  )
}
