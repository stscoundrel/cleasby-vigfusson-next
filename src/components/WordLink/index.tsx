import Link from 'next/link'
import { DictionaryEntry } from 'lib/services/dictionary';
import styles from './WordLink.module.scss'

interface WordLinkProps{
  data: DictionaryEntry,
  useLowerCase: boolean,
}

export default function WordLink({ data, useLowerCase = true }: WordLinkProps) {
  const { slug, word } = data

  return (
    <Link key={`link${slug}`} href={`/word/${slug}`} className={styles.link} prefetch={false}>
       {useLowerCase ? word.toLowerCase() : word}
     </Link>
  );
}
