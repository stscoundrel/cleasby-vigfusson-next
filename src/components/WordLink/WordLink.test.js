import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import WordLink from './index'

const word = {
  word: 'af-burÃ°r',
  slug: 'af-burdr',
}

describe('WordLink component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<WordLink data={word} />)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<WordLink data={word} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Has correct label', () => {
    const tree = renderer.create(<WordLink data={word} />)
    const { root } = tree

    expect(root.findByType('a').children).toEqual([word.word])
  })
})
