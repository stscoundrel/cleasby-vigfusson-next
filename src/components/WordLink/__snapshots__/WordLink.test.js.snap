// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordLink component Matches snapshot 1`] = `
<a
  className="link"
  href="/word/af-burdr"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  af-bur√∞r
</a>
`;
